### PR TITLE
Update canal to v3.9

### PR DESCRIPTION
**Pull request by jascott1**
This replaces previous canal v3.3.

Signed-off-by: Justin Scott <justin.a.scott@intel.com>

### CODE REVIEW

File: clr-k8s-examples/create_stack.sh [hunk 1]
```diff
@@ -279,7 +289,310 @@ function set_repo_version() {
 	popd
 
 }
+function detect_k8s() {
+   if [[ -z "$K8S_VER" ]]; then
+		 K8S_VER=$(kubectl version | grep Server | grep -Po 'GitVersion:"(.*?)",' | grep -Po '"(.*?)"' | tr -d '"')
+	 fi
+}
+detect_k8s
+declare -A command_handlers#!/usr/bin/env bash
```
Comment (hnanni) [inline clr-k8s-examples/create_stack.sh]: + ./create_stack.sh all
./create_stack.sh: line 298: declare: `command_handlers#!/usr/bin/env': not a valid identifier

Comment (jascott1) [reply clr-k8s-examples/create_stack.sh]: Thanks @hnanni It is still a work in progress but almost there.

Comment (jascott1) [reply clr-k8s-examples/create_stack.sh]: @hnanni Updated and removed that call (which was an error) but still working on it overall.

**Comment by jascott1** [comment]
This is ready for review.

File: clr-k8s-examples/create_stack.sh [hunk 2]
```diff
@@ -89,14 +99,19 @@ function kata() {
...
+		CANAL_URL="https://docs.projectcalico.org/v3.3/getting-started/kubernetes/installation/hosted/canal"
+	fi
 	CANAL_DIR="0-canal"
 
 	# canal manifests are not kept in repo but in docs site so use curl
 	mkdir -p "${CANAL_DIR}/overlays/${CANAL_VER}/canal"
 	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/canal.yaml" "$CANAL_URL/canal.yaml"
-	curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/rbac.yaml" "$CANAL_URL/rbac.yaml"
+	if [[ "$CANAL_VER" == "v3.3" ]]; then
+  		curl -o "${CANAL_DIR}/overlays/${CANAL_VER}/canal/rbac.yaml" "$CANAL_URL/rbac.yaml"
```
Comment (NitinAtIntel) [inline clr-k8s-examples/create_stack.sh]: I assume if we don't have 3.3 then we don't need the rbac yaml?

Comment (jascott1) [reply clr-k8s-examples/create_stack.sh]: That's correct.

**Comment by hnanni** [comment]
I've just tested this merge with k8s 1.16. I got:
> servicemonitor.monitoring.coreos.com/prometheus created                                                               
> unable to recognize "4-kube-prometheus/overlays/v0.1.0": no matches for kind "Deployment" in version "apps/v1beta2"
> unable to recognize "4-kube-prometheus/overlays/v0.1.0": no matches for kind "Deployment" in version "apps/v1beta2"
> unable to recognize "4-kube-prometheus/overlays/v0.1.0": no matches for kind "Deployment" in version "apps/v1beta2"
> unable to recognize "4-kube-prometheus/overlays/v0.1.0": no matches for kind "Deployment" in version "apps/v1beta2"
> unable to recognize "4-kube-prometheus/overlays/v0.1.0": no matches for kind "DaemonSet" in version "apps/v1beta2"

I think I'm missing a PR I need to also apply. Do you know which is it?

**Comment by jascott1** [comment]
@hnanni You need v0.2.0 of prometheus for 1.16. I created this PR, feel free to approve it :)
https://github.com/clearlinux/cloud-native-setup/pull/196

**Pull request merged**